components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -64,20 +74,387 @@ pub struct SplitResult {
     pub tablet_index: u64,
 }
 
+#[derive(Debug)]
+pub struct SplitRegionInitInfo {
+    pub parent_region_id: u64,
+    pub parent_epoch: RegionEpoch,
+    /// Split region
+    pub region: metapb::Region,
+    pub parent_is_leader: bool,
+    pub parent_stat: PeerStat,
+    pub approximate_size: Option<u64>,
+    pub approximate_keys: Option<u64>,
+    /// In-memory pessimistic locks that should be inherited from parent region
+    pub locks: PeerPessimisticLocks,
+    pub last_split_region: bool,
+}
+
+#[derive(Debug)]
+pub struct SplitRegionInitResp {
+    pub parent_epoch: RegionEpoch,
+    pub child_region_id: u64,
+    // FIXME: when it is false
+    pub result: bool,
+}
+
+pub enum RegionSplitMsg {
+    SplitRegionInit(Box<SplitRegionInitInfo>),
+    SplitRegionInitResp(Box<SplitRegionInitResp>),
+}
+
 impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
     pub fn propose_split<T>(
         &mut self,
         store_ctx: &mut StoreContext<EK, ER, T>,
-        mut req: RaftCmdRequest,
+        req: RaftCmdRequest,
     ) -> Result<u64> {
-        validate_batch_split(req.mut_admin_request(), self.region())?;
+        validate_batch_split(req.get_admin_request(), self.region())?;
         let mut proposal_ctx = ProposalContext::empty();
         proposal_ctx.insert(ProposalContext::SYNC_LOG);
         proposal_ctx.insert(ProposalContext::SPLIT);
 
         let data = req.write_to_bytes().unwrap();
         self.propose_with_ctx(store_ctx, data, proposal_ctx.to_vec())
     }
+
+    pub fn on_ready_split_region<T>(
```

Comment:
it will be be better to add the original code's fail point now than later.  
such as 
```
 fail_point!("on_split", self.ctx.store_id() == 3, |_| {});
 fail_point!("on_split_invalidate_locks");
```